`-Wenum-compare` reported in `llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp`

```
/home/user/CLionProjects/llvm-project/llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp: In member function ‘llvm::SDValue llvm::AMDGPUTargetLowering::lowerFEXP10Unsafe(llvm::SDValue, const llvm::SDLoc&, llvm::SelectionDAG&, llvm::SDNodeFlags) const’:
/home/user/CLionProjects/llvm-project/llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp:2974: warning: enumerated mismatch in conditional expression: ‘llvm::AMDGPUISD::NodeType’ vs ‘llvm::ISD::NodeType’ [-Wenum-compare]
 2974 |   const unsigned Exp2Op = VT == MVT::f32 ? AMDGPUISD::EXP : ISD::FEXP2;
```

I am using `gcc (GCC) 14.2.1 20240910` on Manjaro Linux.

--- a/llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp
+++ b/llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp
@@ -2971,7 +2971,8 @@ SDValue AMDGPUTargetLowering::lowerFEXP10Unsafe(SDValue X, const SDLoc &SL,
                                                 SelectionDAG &DAG,
                                                 SDNodeFlags Flags) const {
   const EVT VT = X.getValueType();
-  const unsigned Exp2Op = VT == MVT::f32 ? AMDGPUISD::EXP : ISD::FEXP2;
+  const unsigned Exp2Op = VT == MVT::f32 ? static_cast<unsigned>(AMDGPUISD::EXP)
+                                         : static_cast<unsigned>(ISD::FEXP2);
 
   if (VT != MVT::f32 || !needsDenormHandlingF32(DAG, X, Flags)) {
     // exp2(x * 0x1.a92000p+1f) * exp2(x * 0x1.4f0978p-11f);

